src/bundle/Resources/public/js/scripts/core/collapse.js

```diff
@@ -1,11 +1,123 @@
+    const clickedElementsState = initializeClickedStateArray();
+    const toggleMultiCollapseButton = (btn, changeToCollapseAll) => {
+        const expandAll = btn.querySelector('.ibexa-multi-collapse__toggler-expand');
+        const collapseAll = btn.querySelector('.ibexa-multi-collapse__toggler-collapse');
+
+        if (changeToCollapseAll && collapseAll.classList.contains('d-none')) {
+            collapseAll.classList.toggle('d-none');
+            expandAll.classList.toggle('d-none');
+        } else if (!changeToCollapseAll && expandAll.classList.contains('d-none')) {
+            collapseAll.classList.toggle('d-none');
+            expandAll.classList.toggle('d-none');
+        }
+    };
+    const toggleMultiCollapseIfNeeded = (multiCollapseNode, toggleBtn, tabllLength) => {
+        const allElements = multiCollapseNode.querySelectorAll(MULTI_COLLAPSE_ELEMENT_SELECTOR);
+
+        if (tabllLength === allElements.length || tabllLength === 0) {
+            toggleMultiCollapseButton(toggleBtn, tabllLength === 0);
+        }
+    };
+    const handleCollapseAction = (multiCollapseNode, expandAction) => {
+        const index = clickedElementsState.findIndex(
+            (element) => element.btnName === multiCollapseNode.getAttribute(COLLAPSE_SECTION_BODY_TAG),
+        );
+
+        if (expandAction) {
+            clickedElementsState[index].collapsedElements = [];
+        }
+
+        multiCollapseNode.querySelectorAll('.ibexa-collapse').forEach((collapseNode) => {
+            const isElementCollapsed = collapseNode.classList.contains('ibexa-collapse--collapsed');
+
+            if (expandAction === isElementCollapsed) {
+                bootstrap.Collapse.getOrCreateInstance(collapseNode.querySelector(MULTI_COLLAPSE_ELEMENT_SELECTOR)).toggle();
+
+                if (!expandAction) {
+                    const uniqueName = collapseNode.querySelector('.ibexa-collapse__toggle-btn').getAttribute('data-bs-target');
+                    clickedElementsState[index].collapsedElements.push(uniqueName);
+                }
+            }
+        });
+    };
+
     doc.querySelectorAll('.ibexa-collapse').forEach((collapseNode) => {
         const toggleButton = collapseNode.querySelector('.ibexa-collapse__toggle-btn');
         const isCollapsed = toggleButton.classList.contains('collapsed');
 
         collapseNode.classList.toggle('ibexa-collapse--collapsed', isCollapsed);
         collapseNode.dataset.collapsed = isCollapsed;
 
+        if (toggleAllBtns && toggleAllBtns.length > 0) {
+            const multicollapseNode = collapseNode.closest(`[${COLLAPSE_SECTION_BODY_TAG}]`);
+
+            if (!!multicollapseNode) {
+                const uniqueName = toggleButton.getAttribute('data-bs-target');
+                const currentToggleAllButton = Array.from(toggleAllBtns).find(
+                    (button) => button.getAttribute(MULTI_COLLAPSE_BTN_TAG) === multicollapseNode.getAttribute(COLLAPSE_SECTION_BODY_TAG),
+                );
```

Comment:
I think that creating selector for this may be simpler? WDYT? 🤔  (here and in other places)
We also prefer to use `dataset` instead of `getAttribute` for _data-_ attributes.
```suggestion
                const toggleAllBtnId = multicollapseNode.dataset.multiCollapseBody;
                const currentToggleAllButton = doc.querySelector(`[data-multi-collapse-btn-id="${toggleAllBtnId}"]`);
```